greenhouse/main.go

```diff
@@ -69,15 +70,57 @@ var diskCheckInterval = flag.Duration("disk-check-interval", time.Second*30,
 var remount = flag.Bool("remount", false,
 	"attempt to remount --dir with strictatime,lazyatime to improve eviction")
 
+// prometheusMetrics are served by /prometheus on the metrics port
+type prometheusMetrics struct {
+	DiskStats  map[string]prometheus.Gauge
+	CacheStats map[string]prometheus.Counter
```

Comment:
I think it would be more robust to use structs here instead of maps. The keys in these maps are constant so they could just be struct fields. This would validate field names at compile time instead of runtime.

Reply:
I'll do this in a follow up, need to fix it in boskos as well which I admittedly cargo-culted a bit.

Reply:
I'd guess @krzyzacy did this in boskos because it makes registering all the metrics so clean :man_shrugging: